sandboxnu/graduatenu
# Majors - Concentration Support

# Description

https://user-images.githubusercontent.com/30478978/205423941-4b8e7b37-8630-43e1-bc04-ddf6d3e7fe47.mov

Adds support for concentrations + fixes to the bscs major.

We can now, 
- create a valid plan with a valid concentration
- update a plan with another valid concentration
- view selected concentration section validated in the sidebar
- view selected concentration title in the sidebar

## Changes
- Return supported concentrations in support majors API endpoint
- Use and display supported concentrations in add/edit plan 
- Validate concentration in update/create plan API endpoints
- Display selected concentration section and validate it
- Display selected concentration title in sidebar
- Fix BSCS errors

## Type of change

Please delete options that are not relevant.


# How Has This Been Tested?

Frontend: manually, see video.
Backend: manually, postman.

- Create, no concentration when concentration required
 <img width="1439" alt="create-invalid-no-concentration" src="https://user-images.githubusercontent.com/30478978/205423263-191df28c-6004-4d7c-88c5-ca3f287f6624.png">

- Create, invalid concentration
<img width="1440" alt="create-invalid-concentration" src="https://user-images.githubusercontent.com/30478978/205423267-c38a3c8b-f891-46b5-88b8-49c24244621c.png">

- Create, valid con
<img width="1440" alt="create-valid-concentration" src="https://user-images.githubusercontent.com/30478978/205423300-3d3ed9cf-9624-447b-9dca-406cb2601019.png">
centration

- Update, invalid concentration
<img width="1440" alt="update-invalid-concentration" src="https://user-images.githubusercontent.com/30478978/205423255-5752b617-9b5e-4fe1-a52d-82cc7be63b05.png">

- Update, valid concentration 
<img width="1440" alt="update-valid-concentration" src="https://user-images.githubusercontent.com/30478978/205423258-9e109793-f04a-46c3-a955-ff39e78a956f.png">

# Checklist:
